From 98b0088a97f55e8f36a8e4cac70129d9d1d69a7e Mon Sep 17 00:00:00 2001
From: Hari Subramaniyan <harisankar@apache.org>
Date: Wed, 1 Jun 2016 16:57:00 -0800
Subject: [PATCH 0850/1164] HIVE-13911 : load inpath fails throwing
 org.apache.hadoop.security.AccessControlException
 (Hari Sankar Subramaniyan via Ashutosh Chauhan)

Signed-off-by: Ashutosh Chauhan <hashutosh@apache.org>
(cherry picked from commit b7de5b81b32ff901d23e6241a64f204622d3c6af)

Change-Id: Ie326aafc76e5d2a90aa86baa958b2d04f5c7c163
---
 .../org/apache/hadoop/hive/ql/metadata/Hive.java   |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java b/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
index c4fc7d3..a56a477 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java
@@ -2527,7 +2527,8 @@ private static void copyFiles(final HiveConf conf, final FileSystem destFs,
         // If we do a rename for a non-local file, we will be transfering the original
         // file permissions from source to the destination. Else, in case of mvFile() where we
         // copy from source to destination, we will inherit the destination's parent group ownership.
-        final String srcGroup = srcFile.getGroup();
+        final String srcGroup = isRenameAllowed ? srcFile.getGroup() :
+          fullDestStatus.getFileStatus().getGroup();
         if (null == pool) {
           try {
             Path destPath = mvFile(conf, srcFs, srcP, destFs, destf, isSrcLocal, isRenameAllowed);
-- 
1.7.9.5

